src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -5052,3 +5052,199 @@ TEST_CASE("GradientMultiheadAttentionTest", "[ANNLayerTest]")
 
   REQUIRE(CheckGradient(function) <= 3e-06);
 }
+/**
+ * Simple tests for instance normalization layer.
+ */
+TEST_CASE("InstanceNormLayerTest", "[ANNLayerTest]")
+{
+  arma::mat input, result, output, delta, deltaExpected;
+  arma::mat runningMean, runningVar;
+
+  // Represents 2 images, each having 3 channels, and shape (3,2).
+  input << 1  << 19  << arma::endr
+        << 2  << 20  << arma::endr
+        << 3  << 21  << arma::endr
+        << 4  << 22  << arma::endr
+        << 5  << 23  << arma::endr
+        << 6  << 24  << arma::endr
+        << 7  << 25  << arma::endr
+        << 8  << 26  << arma::endr
+        << 9  << 27  << arma::endr
+        << 10 << 28  << arma::endr
+        << 11 << 29  << arma::endr
+        << 12 << 30  << arma::endr
+        << 13 << 31  << arma::endr
+        << 14 << 32  << arma::endr
+        << 15 << 33  << arma::endr
+        << 16 << 34  << arma::endr
+        << 17 << 35  << arma::endr
+        << 18 << 36  << arma::endr;
+
+  // Output calculated using torch.nn.InstanceNorm2d().
```

Comment:
Do you have the code for that one at hand?

Reply:
[Google Colab notebook](https://colab.research.google.com/drive/18E8a-LTHFOpqa6w53vOsjRz7nbRHLvaf?usp=sharing)